internal/pkg/mapper/configmetadata/remote_test.go

```diff
@@ -87,8 +106,36 @@ func TestConfigMetadataMapAfterRemoteLoad(t *testing.T) {
 	changes := model.NewRemoteChanges()
 	changes.AddLoaded(configState)
 	changes.AddLoaded(configState2)
-
 	assert.NoError(t, mockedState.Mapper().OnRemoteChange(changes))
 	assert.Equal(t, map[string]string{"KBC.KaC.Meta": "value1", "KBC.KaC.Meta2": "value2"}, config.Metadata)
 	assert.Equal(t, make(map[string]string), config2.Metadata)
 }
+
+func TestConfigMetadataOnRemoteChangeSaved(t *testing.T) {
+	t.Parallel()
+	mockedState := InitState(t)
+
+	content := orderedmap.New()
+	json.MustDecodeString("{}", content)
+
+	// Config with metadata
+	configKey := model.ConfigKey{
+		BranchId:    123,
+		ComponentId: "keboola.ex-aws-s3",
+		Id:          "456",
+	}
+	configManifest := &model.ConfigManifest{
+		ConfigKey: configKey,
+	}
+	config := &model.Config{ConfigKey: configKey, Content: content, Metadata: map[string]string{"KBC.KaC.meta1": "val1"}}
+	configState := &model.ConfigState{
+		ConfigManifest: configManifest,
+		Remote:         config,
+	}
+	assert.NoError(t, mockedState.Set(configState))
+
+	// Invoke
+	changes := model.NewRemoteChanges()
+	changes.AddSaved(configState)
+	assert.NoError(t, mockedState.Mapper().OnRemoteChange(changes))
```

Comment:
Tu by bolo dobre zvalidvoat, ze sa request poslal, najdi si v kode `httpTransport.GetCallCountInfo()`.

Reply:
👍 paráda, to jsem nevěděl že jde